docs/02-providers/aws/01-setup.md

```diff
@@ -69,37 +69,132 @@ Default output format [None]: ENTER
+One set of credentials for all stages using environment variables
+```bash
+export AWS_ACCESS_KEY_ID=<key>
+export AWS_SECRET_ACCESS_KEY=<secret>
+export AWS_SESSION_TOKEN=<token>
+serverless <...>
+```
+
+A set of credentials for each stage using environment variables
+```bash
+export AWS_TEST_ACCESS_KEY_ID=<key>
+export AWS_TEST_SECRET_ACCESS_KEY=<secret>
+export AWS_TEST_SESSION_TOKEN=<token>
+
+export AWS_PROD_ACCESS_KEY_ID=<key>
+export AWS_PROD_SECRET_ACCESS_KEY=<secret>
+export AWS_PROD_SESSION_TOKEN=<token>
+
+serverless <...>
+```
+
+A profile for all stages using environment variables
+```bash
+export AWS_PROFILE=<profile>
+serverless <...>
+```
 
+A profile for each stage using environment variables
+```bash
+export AWS_TEST_PROFILE=<profile>
+
+export AWS_PROD_PROFILE=<profile>
+
+serverless <...>
+```
+
+#### Credential & Profile Overriding
+
+Sometimes you want to be able to specify a default but to override that default for a special case.  This is possible with credentials and profiles in Serverless.  You may specify credentials and profiles in various forms.  The serverless.yml has the lowest priority and environment variables used for all stages will override values set in serverless.yml.  Environment variables that are specific to a stage have the highest priority and will override both broad environment variables as well as serverless.yml.  Profile provided credentials will override credentials provided in piece-meal from otherwise equivalent credential sources.  A priority listing follows.
+
+severless.yml credentials < serverless.yml profile credentials < all-stages environment credentials < all stages environment profile credentials < stage-specific environment credentials < stage-specific environment profile credentials
+
+A default set of `prod` credentials to use overriden by stage specific credentials
+```bash
+export AWS_ACCESS_KEY_ID=<key>
+export AWS_SECRET_ACCESS_KEY=<secret>
+export AWS_SESSION_TOKEN=<token>
+
+export AWS_PROD_ACCESS_KEY_ID=<prod-key>
+export AWS_PROD_SECRET_ACCESS_KEY=<prod-secret>
+export AWS_PROD_SESSION_TOKEN=<prod-token>
+
+serverless <...>
+```
+
+A default profile to use overriden by a `prod` specific profile
+```bash
+export AWS_PROFILE=<profile>
+
+export AWS_PROD_PROFILE=<profile>
+
+serverless <...>
+```
+
+A default profile declared in serverless.yml overridden by a `prod` specific environment variable profile
 ```yml
-service: new-service
 provider:
-  name: aws
-  runtime: nodejs4.3
-  stage: ${opt:stage, self:custom.defaultStage}
-  profile: ${self:custom.profiles.${self:provider.stage}}
-custom:
-  defaultStage: dev
-  profiles:
-    dev: devProfile
-    prod: prodProfile
+  profile: your-profile
+```
+```bash
+export AWS_PROD_ACCESS_KEY_ID=<prod-key>
+export AWS_PROD_SECRET_ACCESS_KEY=<prod-secret>
+export AWS_PROD_SESSION_TOKEN=<prod-token>
+
+serverless <...>
 ```
 
+Et cetera
```

Comment:
this and prior implemented here: https://github.com/serverless/serverless/pull/2297